content/docs/iac/using-pulumi/crossguard/core-concepts.md

```diff
@@ -90,20 +90,29 @@ There are no restrictions on which policies you combine within a pack, and you s
+1. **Resource Policies:** These validate a particular resource in a stack or account before the resource is created or updated, looking at the resource's *input* properties.
+1. **Stack Policies:** These validate all resources in the stack after the Pulumi operation has processed all resources, looking at each resource's *output* properties. Stack policies are most useful during preview operations.
 
 This table summarizes the primary differences between the two types:
 
 |                                | Resource Policies                     | Stack Policies                                                                        |
 |--------------------------------|---------------------------------------|---------------------------------------------------------------------------------------|
 | What does it check?            | Individual resources                  | All resources in the stack                                                            |
-| When is the check performed?   | Before resources are created/modified | After all stack resources have been created/modified                                  |
+| When is the check performed?   | Before resources are created/modified | After all stack resources have been created/modified (see note below)               |
 | Can it remediate?              | Yes                                   | No                                                                                    |
-| What information is available? | Resource _input_ properties           | Resource _output_ properties (Note: inputs are propagated to outputs during preview)  |
+| What information is available? | Resource *input* properties           | Resource *output* properties (Note: inputs are propagated to outputs during preview)  |
 | What is the type name?         | `ResourceValidationPolicy`            | `StackValidationPolicy`                                                               |
 | Supported for Insights?        | Yes                                   | No - IaC-specific feature                                                             |
 
+{{% notes type="info" %}}
+**Stack Policy Timing**: The timing of stack policy execution differs between `pulumi preview` and `pulumi update`:
+
+- During `pulumi preview`: Stack policies run after the preview completes, using whatever output values are already known from previously provisioned resources. Since no new resources are actually created during preview, stack policies are most useful in this context for catching violations before any provisioning occurs.
+- During `pulumi update`: Stack policies run after all resources have been provisioned, but before the update operation completes. This means non-compliant resources may already be created by the time the policy violation is detected.
+
+For this reason, stack policies are most effective when used with `pulumi preview` to identify issues before actual resource provisioning.
```

Comment:
I think there are some subtle tradeoffs here. Preview won't necessarily resolve all outputs, so the policy may be evaluating a different set of properties depending on the operation. However, as you note, preview gates all changes to the stack.

Reply:
Can you give me an example of where an output might not be resolved in a preview if a resource was already provisioned in a `pulumi up`? (That's what I was trying to communicate there, so if that was not clear I will reword it.)